arch/arm/core/aarch32/cortex_m/mpu/arm_core_mpu.c

```diff
@@ -148,6 +148,20 @@ void z_arm_configure_dynamic_mpu_regions(struct k_thread *thread)
 	 * of the respective dynamic MPU regions to be programmed for
 	 * the given thread. The array of partitions (along with its
 	 * actual size) will be supplied to the underlying MPU driver.
+	 *
+	 * The drivers of what regions get configured are CONFIG_USERSPACE,
+	 * CONFIG_MPU_STACK_GUARD, and K_USER/supervisor threads.
+	 *
+	 * If CONFIG_USERSPACE then any Memory Domains associated with the
+	 * thread get a defined region.
+	 *
+	 * If CONFIG_USERSPACE and the thread is a K_USER thread the usermode
+	 * thread stack is defined a region.
+	 *
+	 * IF CONFIG_MPU_STACK_GUARD is defined the thread is a privilege
```

Comment:
... **and** the thread is a supervisor thread

Reply:
okay. But one inconsistency I ran across is that we use "priv_stack_start" in the name for the supervisor stack. Bugs me for some reason and part of the reason I changed up my language here.

Reply:
it's called that because it is used as the stack for temporary CPU privilege level elevations when handling syscalls from user mode.

Reply:
thanks... was confusing to me when first looking at this but I understand now